airflow/providers/amazon/aws/hooks/opensearch.py

```diff
@@ -0,0 +1,130 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+from typing import Any
+
+from opensearchpy import AWSV4SignerAuth, OpenSearch, RequestsHttpConnection
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class OpenSearchHook(AwsBaseHook):
+    """
+    This Hook provides a thin wrapper around the OpenSearch client.
+
+    :param: open_search_conn_id: AWS Connection to use with Open Search
+    :param: log_query: Whether to log the query used for Open Search
+    """
+    conn_name_attr = "opensearch_conn_id"
+    default_conn_name = "opensearch_default"
+    conn_type = "opensearch"
+    hook_name = "AWS Open Search Hook"
+
+    def __init__(self, *args: Any, open_search_conn_id: str, log_query: bool, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.conn_id = open_search_conn_id
+        self.log_query = log_query
+
+        conn = self.get_connection(self.conn_id)
+        self.use_ssl = conn.extra_dejson.get("use_ssl", False)
+        self.verify_certs = conn.extra_dejson.get("verify_certs", False)
+
+        self.__SERVICE = "es"
+        self._credentials = self.get_credentials(self.region_name)
+        self._auth = AWSV4SignerAuth(self._credentials, self.region_name, self.__SERVICE)
+
+        self.client = OpenSearch(
+            hosts=[{"host": conn.host, "port": conn.port}],
+            http_auth=self._auth,
+            use_ssl=self.use_ssl,
+            verify_certs=self.verify_certs,
+            connection_class=RequestsHttpConnection,
+        )
```

Comment:
We should avoid to call expensive call in Hook constructor, this might be move into the cached properties

Reply:
Moved into the get client method as a cached property.